### PR TITLE
Optimizations and refactor cards

### PR DESCRIPTION
**Pull request by LHolten**
Cards are now defined really compactly, values are copied from Alion.
If this is not Ok, I can try to change back to the human readable form.
The biggest optimization is the switch to arrayvec.

### CODE REVIEW

File: onitama/src/cards.rs [hunk 1]
```diff
@@ -379,19 +212,11 @@ mod tests {
     #[test]
     fn test_reverse_bitmap() {
         assert_eq!(
-            reverse_bitmap(Card::Eel.get_moves()),
-            Card::Cobra.get_moves()
+            Card::Eel.get_white().into_value(),
+            Card::Cobra.get_black().into_value()
         )
```
Comment (ViliamVadocz) [inline onitama/src/cards.rs]: This doesn't seem to use reverse_bitmap anymore, yet it's supposed to be testing it.

File: onitama/src/cards.rs [hunk 2]
```diff
@@ -26,209 +27,42 @@ pub enum Card {
...
-                       0 1 0 0 0
-                       0 0 0 0 0),
-            Card::Tiger =>
-                board!(0 0 1 0 0
-                       0 0 0 0 0
-                       0 0 0 0 0
-                       0 0 1 0 0
-                       0 0 0 0 0),
-        }
+    pub fn get_white(self) -> Bitmap<U25> {
```
Comment (ViliamVadocz) [inline onitama/src/cards.rs]: I like the name change

File: onitama/src/cards.rs [hunk 3]
```diff
@@ -26,209 +27,42 @@ pub enum Card {
...
+        Card::Mantis => 0b00000_01010_00000_00100_00000,
+        Card::Boar => 0b00000_00100_01010_00000_00000,
+        Card::Frog => 0b00000_01000_10000_00010_00000,
+        Card::Rabbit => 0b00000_00010_00001_01000_00000,
+        Card::Goose => 0b00000_01000_01010_00010_00000,
+        Card::Rooster => 0b00000_00010_01010_01000_00000,
+        Card::Horse => 0b00000_00100_01000_00100_00000,
+        Card::Ox => 0b00000_00100_00010_00100_00000,
+        Card::Eel => 0b00000_01000_00010_01000_00000,
+        Card::Cobra => 0b00000_00010_01000_00010_00000,
```
Comment (ViliamVadocz) [inline onitama/src/cards.rs]: This is hard to read. I prefer the human-readable macro.

File: onitama/src/cards.rs (hunk 1 above)
Comment (LHolten) [reply onitama/src/cards.rs]: the black cards a created by reversing the white cards, so this checks the precomputed result

**Pull request merged**